### PR TITLE
fix: BaseBuilder::getCompiledDelete() runs real query

### LINKED ISSUE #4180: Bug: BaseBuilder::getCompiledDelete() runs real query
**Describe the bug**
`BaseBuilder::getCompiledDelete()` runs real query.

https://github.com/codeigniter4/CodeIgniter4/blob/7ca2a7193ae0a109c75fd47233ae954a6db350e0/system/Database/BaseBuilder.php#L2808

**CodeIgniter 4 version**
develop branch.

**Affected module(s)**
Database

**Expected behavior, and steps to reproduce if appropriate**
Compiles the deletion query, but does not run the query.

**Context**
- OS: [macOS 10.15.7]
- Web server [PHP built-in]
- PHP version [7.4.13]

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes  #4180

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by iRedds — APPROVED** [review]

File: system/Database/BaseBuilder.php [hunk 1]
```diff
@@ -2803,9 +2803,9 @@ protected function _truncate(string $table): string
 	 */
 	public function getCompiledDelete(bool $reset = true): string
 	{
-		$table = $this->QBFrom[0];
-
-		$sql = $this->delete($table, null, $reset);
+		$this->testMode();
+		$sql = $this->delete('', null, $reset);
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2807]:
```suggestion
		$sql = $this->testMode()->delete('', null, $reset);
```

File: tests/system/Database/Builder/DeleteTest.php [hunk 2]
```diff
@@ -35,4 +35,28 @@ public function testDelete()
...
 
+	public function testGetCompiledDelete()
+	{
+		$builder = $this->db->table('jobs');
+
+		$builder->where('id', 1);
+		$sql = $builder->getCompiledDelete();
+
+		$expectedSQL = 'DELETE FROM "jobs"'
+			. "\n" . 'WHERE "id" = 1';
```
Comment (paulbalandan) [inline tests/system/Database/Builder/DeleteTest.php:46]:
```suggestion
		$expectedSQL = "DELETE FROM \"jobs\"\n WHERE \"id\" = 1";
```

File: tests/system/Database/Builder/DeleteTest.php [hunk 3]
```diff
@@ -35,4 +35,28 @@ public function testDelete()
...
+	public function testGetCompiledDeleteWithLimit()
+	{
+		$builder = $this->db->table('jobs');
+
+		$builder->where('id', 1);
+		$builder->limit(10);
+		$sql = $builder->getCompiledDelete();
+
+		$expectedSQL = 'DELETE FROM "jobs"'
+			. "\n" . 'WHERE "id" = 1 LIMIT 10';
```
Comment (paulbalandan) [inline tests/system/Database/Builder/DeleteTest.php:59]:
```suggestion
		$expectedSQL = "DELETE FROM \"jobs\"\n WHERE \"id\" = 1 LIMIT 10";
```

File: tests/system/Database/Builder/DeleteTest.php [hunk 4]
```diff
@@ -35,4 +35,28 @@ public function testDelete()
...
+		$this->assertEquals($expectedSQL, $sql);
+	}
+
+	public function testGetCompiledDeleteWithLimit()
+	{
+		$builder = $this->db->table('jobs');
+
+		$builder->where('id', 1);
+		$builder->limit(10);
+		$sql = $builder->getCompiledDelete();
```
Comment (paulbalandan) [inline tests/system/Database/Builder/DeleteTest.php:56]:
```suggestion
		$sql = $builder->where('id', 1)->limit(10)->getCompiledDelete();
```

File: system/Database/BaseBuilder.php [hunk 5]
```diff
@@ -2803,9 +2803,9 @@ protected function _truncate(string $table): string
 	 */
 	public function getCompiledDelete(bool $reset = true): string
 	{
-		$table = $this->QBFrom[0];
-
-		$sql = $this->delete($table, null, $reset);
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2808]: Forgive me for being dumb but why is the `$table` removed?

Comment (iRedds) [reply system/Database/BaseBuilder.php:2808]: The method expects condition on first argument.
https://github.com/codeigniter4/CodeIgniter4/blob/7ca2a7193ae0a109c75fd47233ae954a6db350e0/system/Database/BaseBuilder.php#L2827-L2834

Comment (paulbalandan) [reply system/Database/BaseBuilder.php:2808]: Ah yes. See it now. Thanks for that.

File: tests/system/Database/Builder/DeleteTest.php (hunk 2 above)
Comment (kenjis) [reply tests/system/Database/Builder/DeleteTest.php:46]: It seems the space before `WHERE` is not needed.
And isn't it difficult to read `"DELETE FROM \"jobs\"\nWHERE \"id\" = 1"`?

Comment (paulbalandan) [reply tests/system/Database/Builder/DeleteTest.php:46]: Yes, it's quite difficult. Maybe we can use `str_replace` just like in the previous test case?

Comment (kenjis) [reply tests/system/Database/Builder/DeleteTest.php]: I pushed a commit. How about this?

Comment (paulbalandan) [reply tests/system/Database/Builder/DeleteTest.php]: It's good to me.

**Review by paulbalandan — APPROVED** [review]

**Review by iRedds — APPROVED** [review]

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @kenjis

**Comment by diannurdiana86** [comment]
Thank you @kenjis

**Pull request merged**